crates/sema/src/ast_passes.rs

```diff
@@ -98,6 +99,55 @@ impl<'sess> AstValidator<'sess, '_> {
             report("remove underscores in front of the exponent");
         }
     }
+
+    fn check_subdenominations_for_number_literals(
+        &self,
+        lit: &ast::Lit,
+        subdenomination: &Option<ast::SubDenomination>,
+    ) {
+        match lit.kind {
```

Comment:
can let-else/matches! at the top for early returning if not `Number(_)`; also should include `Rational` since `Number` only covers integers

Reply:
ok done